app/lib/clients/vault/oidc.rb

```diff
@@ -0,0 +1,151 @@
+=begin
+
+The purpose of this module is to assign a policy to an OIDC user, by
+mapping that user's email address to a policy we create.
+It works as follows:
+
+It creates an OIDC provider and user.  That user has a
+username/password/email addr, that can be accessed with OIDC auth.
+
+It creates an OIDC client which connects to that provider.  When a
+user tries to auth, the client connects to the provider, which opens
+up a browser window allowing the user to enter his username/password.
+
+On success, the provider returns an OIDC token, which includes the
+user's email addr.
+
+The client has been configured to map that email address to an entity
+in vault, which has the policy which we want the user to have.
+
+So the mapping goes from the email address on the provider, to the
+policy in vault.  The email addr may not be the best mapping to use.
+Some other piece of user info may ultimately be better used to map the
+user to the policy.  But we don't yet have a good understanding of
+different OIDC provider configurations, so this should be good enough
+for now
+
+Note that this provider is only meant to be used in our dev/test
+environment to excercise the client.  In a prod env, a real OIDC
+provider is configured in.
+
+=end
+module Clients
+  class Vault
+    module Oidc
+      def configure_oidc_provider
+        create_provider_webapp
+        create_provider_with_email_scope
+        create_entity_for_initial_user
+        create_userpass_for_initial_user
+        map_userpass_to_entity
+      end
+
+      def configure_oidc_client(issuer, client_id, client_secret)
+        create_client_config(issuer, client_id, client_secret)
+        create_default_policy_for_role
+        create_default_role(client_id)
+      end
+
+      def configure_oidc_user(name, email, policy)
+        client.sys.put_policy(email, policy)
+        put_entity(name, email)
+        put_entity_alias(name, email, "oidc")
+      end
+
+      private
+      cattr_accessor :client_id
+      cattr_accessor :client_secret
+      WEBAPP_NAME = "identity/oidc/client/astral"
+
+      def oidc_provider
+        ::Vault::Client.new(
+          address: "http://oidc_provider:8300",
```

Comment:
Should this be parameterized?

Reply:
It is never going to change because this provider is only used in the docker dev/test environments.  If I add a comment to that effect, so it doesn't confuse reviewers, would that be sufficient.

I don't really want to make a config parameter, because it is not intended to be configured.

Reply:
changed to config param in astral.yml